pkg/apis/trainer/v1alpha1/trainjob_types.go

```diff
@@ -114,7 +114,7 @@ type TrainJobSpec struct {
 
 	// Custom overrides for the training runtime.
 	// +listType=atomic
-	PodSpecOverrides []PodSpecOverride `json:"podSpecOverrides,omitempty"`
+	PodTemplateOverrides []PodTemplateOverride `json:"podTemplateOverrides,omitempty"`
```

Comment:
Since we already released PodSpecOverrides API, we should make it as deprecated and remove in the future version (e.g. v1beta1 or v1alpha2). WDYT @xigang @tenzen-y @astefanutti ?

Reply:
Right, that would make sense to introduce the change in a newer version of the API, either v1beta1 or v1alpha2.

According to https://kubernetes.io/docs/reference/using-api/#api-versioning, it may not really be needed to mark this as deprecated in the current v1alpha1 version.

Reply:
Sorry for the confusion @xigang.

We talked offline with @astefanutti and @tenzen-y, and we decided to remove `PodSpecOverrides` API in this PR as you proposed. We don't need to provide backward compatibility for that API in alpha.

If we have any concerns what that before Trainer v2.1 release, we will bump API to v1alpha2 or v1beta1.